src/go/rpk/pkg/cli/registry/schema/schema.go

```diff
@@ -137,6 +138,19 @@ func typeFromFile(schemaFile string) (sr.SchemaType, error) {
 	}
 }
 
+// This regexp captures the reference flag in its simplest form which is:
+//
+//	name:subject:version
+//
+// However, name can also be a URI. Therefore, this regexp has 3 capturing
+// groups:
+//  1. Either alphabetic characters (scheme) followed by '://', then followed by
+//     any other character except ':', or ':' followed by numbers and more
+//     characters to capture optional ports in the authority and the path.
+//  2. Captures any character except ':' (subject).
+//  3. Captures any character except ':' (version).
+var referenceRegex = regexp.MustCompile(`^([a-zA-Z]+://[^:]+(?::\d+)?(?:/[^:]+)*|[^:]+):([^:]+):([^:]+)$`)
```

Comment:
I could validate here that the third capturing group is a digit, but the error from L174 (strconv.Atoi) is better for the user to understand where the problem is.